### ISSUE TITLE
Can't connect to primary when secondary is down

### ISSUE THREAD

**Issue by Lisio** [mongodb/mongo-php-driver#573]
### Environment

Ubuntu 16.04
PHP 7.1.3
mongodb 1.2.8

### Description

I have cluster with 2 data nodes and 1 arbiter:
mongo-1.domain.com - secondary
mongo-2.domain.com - primary
mongo-a.domain.com - arbiter

When the secondary is down I receive error "No suitable servers found (`serverSelectionTryOnce` set): [connection refused calling ismaster on 'mongo-2.domain.com:27017']"

```
new Manager('mongodb://username:password@mongo-2.domain.com:27017/dbname?replicaSet=test', [
    'slaveOk' => false,
    'readPreference' => 'primary',
]);
```

If I remove "?replicateSet=test" part from dsn string it connects with no problem. 

```
test:PRIMARY> rs.status().members[1]
{
    "_id" : 1,
    "name" : "mongo-2.domain.com:27017",
    "health" : 1,
    "state" : 1,
    "stateStr" : "PRIMARY",
    "uptime" : 107832,
    "optime" : {
        "ts" : Timestamp(1491491348, 1),
        "t" : NumberLong(17)
    },
    "optimeDate" : ISODate("2017-04-06T15:09:08Z"),
    "electionTime" : Timestamp(1491488639, 2),
    "electionDate" : ISODate("2017-04-06T14:23:59Z"),
    "configVersion" : 547206,
    "self" : true
}
```

Why driver doesn't work if the only server it tries to connect to is primary and arbiter is available?

**Comment by derickr** [comment]
This *seems* expected.

In your connection string, you only specify one host (``mongo-2.domain.com:27017``), which the driver says gives a "connection refused".

We would recommend you add both non-arbiter nodes to your connection string.

**Comment by Lisio** [comment]
mongo-2.domain.com - is a **working primary** node. I expect that driver will work with this primary node.

Nothing changes when I add secondary to connection string, except that now I see 2 errors.
```
new Manager('mongodb://username:password@mongo-2.domain.com:27017,mongo-1.domain.com/dbname?replicaSet=test', [
    'slaveOk' => false,
    'readPreference' => 'primary',
]);
```

> No suitable servers found (`serverSelectionTryOnce` set): [connection refused calling ismaster on 'mongo-2.domain.com:27017'] [connection refused calling ismaster on 'mongo-1.domain.com:27017']

**Comment by jmikola** [comment]
> No suitable servers found (serverSelectionTryOnce set): [connection refused calling ismaster on 'mongo-2.domain.com:27017'] [connection refused calling ismaster on 'mongo-1.domain.com:27017']

According to this error, the driver cannot communicate with either the primary or secondary node.

Feel free to follow up with [`mongodb.debug`](http://php.net/manual/en/mongodb.configuration.php) trace logs for all three scenarios (primary without `replicaSet` option, primary with `replicaSet` option, primary and secondary with `replicaSet` option), and we may have a better understanding of what's happening internally.

On a side note, I would suggest removing the `slaveOk` option, as it is deprecated in light of `readPreference`. Also, `readPreference=primary` is the default, so that may be redundant as well.

**Comment by Lisio** [comment]
@jmikola thanks for mongodb.debug link, it helped to solve the issue.
@derickr thank you for your help

I failed twice. The first fail is that mongo-2.domain.com works on the same machine where code was executed, but in container, and I had two lines in /etc/hosts: for 127.0.0.1 and for 192.168.0.10. I removed 127.0.0.1 and everything works as expected.
Second fail was that in fact I used 192.168.0.10 without "replicaSet=test" and mongo-2.domain.com with "replicaSet=test" and misinformed you.
